src/poliastro/util.py

```diff
@@ -84,3 +84,321 @@ def alinspace(start, stop=None, *, num=50, endpoint=True):
         alinspace_fast(start.to_value(u.rad), stop.to_value(u.rad), num, endpoint)
         * u.rad
     )
+
+
+def variable_step_bisection_algorithm(
+    f,
+    x0,
+    y,
+    step_size,
+    args=(),
+    kwargs={},
+    xtol=None,
+    xrtol=None,
+    tol=None,
+    rtol=None,
+    maxiter=None,
+):
+
+    """It calculates the closest value x that approximate the function f to the target value.
+    This calculation process start with the initial value and tries to reduce the error, |y - f(x)|, by adding
+    a steps ize to x in the direction that reduces the error.
+    What makes this process interesting is that the step size is variable. How the step size changes is similar to
+    VSLMS (Variable Sign Linear Mean Square) algorithm. The step size increases when the error has the same sign,
+    avoiding multiple unnecessary iterations and when the error changes the sign it reduces the step size to obtain
+    better approximation. The process will end when the error meets the tolerance conditions or when it exceeds
+    the number of iterations.
+
+    Parameters
+    ----------
+    f : function
+        The function whose want to aproximate to y value.
+        It must be a function of a single variable of the form f(x,a,b,c...), where a,b,c...
+        are extra arguments that can be passed in the args and kwargs parameters.
+        It will use the form f(x, *args, **kwargs)
+    x0 : float
+        An initial estimate of x. It's recommended to be near the actual solution to avoid extra computing
+    y : float
+        The value that want to be estimated
+    step_size : float
+        The initial value of the stepsize be added to x in order to reduce the error |y - f(x)|
+    args : tuple, optional
+        Extra arguments to be used in the function call.
+    kwargs : dict, optional
+        Extra arguments to be used in the function call.
+    xtol : float, optional
+        The allowable tolerance of the x.
+    xrtol : float, optional
+        The allowable relative tolerance of the x.
+    tol : float, optional
+        The allowable tolerance of the error, |y - f(x)|.
+    rtol : float, optional
+        The allowable relative tolerance of the error using y as reference.
+    maxiter : int, optional
+        Maximum number of iterations.
+
+    Returns
+    -------
+    x: float
+        The closest obtained value of f to y. f(x) ~= y
+    """
```

Comment:
Any reference of where this algorithm came from?

Reply:
This is explained in the following comment